hrqb/utils/data_warehouse.py

```diff
@@ -0,0 +1,62 @@
+"""hrqb.utils.data_warehouse"""
+
+import logging
+
+import pandas as pd
+from attrs import define, field
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+
+from hrqb.config import Config
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class DWClient:
+    """Client to provide Oracle Data Warehouse connection and querying."""
+
+    connection_string: str = field(
+        factory=lambda: Config().DATA_WAREHOUSE_CONNECTION_STRING,
```

Comment:
Unsure if I commented on this in tests, but using a `factory=lambda: ...` pattern here is helpful for testing.  This ensures that only when the `DWClient` is instantiated, does it pull the env var `DATA_WAREHOUSE_CONNECTION_STRING` if an explicit connection string is not passed.

If this has been `default=Config().DATA_WAREHOUSE_CONNECTION_STRING`, then monkey patching the env var at pytest time would not be sufficient, as this class is already defined with the default value for any future instances being the env var _before_ that monkey patching.

Reply:
[Non-blocking] Maybe pull some of this information into the doc string or as a comment above the declaration of `connection_string`?